enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,531 @@
+- Recover pools from cluster api provider kubevirt pod restarts or node restarts.
+
+### Non-Goals
+
+- Replacing existing network configuration mechanisms in standalone OpenShift clusters
+- Implementing a full-featured IPAM solution with conflict detection across multiple management clusters
+- Providing network configuration for components outside of HyperShift guest cluster nodes
+- Support complex network configurations like bonding of interfaces and others usually configured with ignition
+
+## Proposal
+
+This proposal introduces static IPAM capabilities through modifications to three key components:
+
+1. **Cluster API Provider KubeVirt**: Extend the API to support IP pool definitions and network configuration, implement IP allocation logic from defined pools
+2. **HyperShift**: Add network configuration options to the KubeVirt platform specification at nodepool, implement translation from this network configuration to capk ip pool and openstack network config
+3. **CoreOS Afterburn**: Enable parsing and application of OpenStack network config standard data as dracut network kernel args from cloud-init config drive for KubeVirt VMs, this is similar what is done at proxmoxve provider.
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for deploying a HyperShift hosted cluster.
+
+1. Disable the default pod network and select the network to replace it:
+   - Sets `AttachDefaultNetwork=false` in the NodePool specification
+   - Specifies an additional network attachment that will replace the default network
+   - Ensures the static IP configuration matches the custom network's name
+
+2. The cluster administrator defines an IP pool configuration in the HostedCluster nodepool specification, including:
+   - Network name (to assign a ip pool)
+   - IP address ranges in CIDR notation, hyphenated format, or as individual IPs
+   - Gateway address
+   - DNS nameservers
+
+3. The cluster administrator creates the HostedCluster resource with the KubeVirt platform type and network configuration
+
+4. HyperShift reconciles the kubevirt nodepool and create KubevirtMachineTemplate resource for the guest cluster nodes.
+   - Configure interface `eno101`, gateways, and nameservers in the OpenStack network config data (without addresses) and store at KubevirtMachineTemplate `networkData` field
+   - Pass nodepool's Network and addresses to KubevirtMachineTemplate NeworkName and Subnets
+   - Configure `acpiIndex: 101` on the virtual machine first interface that will have this network so we have interface name `eno101`
+   - Pass interface `eno101` to KubevirtMachineTemplate Interface.
+   - Create the generated KubevirtMachineTemplate.
+
+5. The Cluster API Provider KubeVirt controller reconcile the KubevirtMachineTemplate:
+   - Allocates an available IP address from the network's pool (there are different pools per network)
```

Comment:
this part about multiple pools per network made me wonder. How is a pool chosen ? i.e. which pool from the network will be used ? how does the controller know which pool to use for allocating IPs to VM-A ? And how would VM-ABC get its IP from a different pool ?

Reply:
Let's take a step back - there's something I don't understand.

You can / want to have multiple pools per network. The first question is why ? Upgrade reasons ? i.e. the pool exhausted, so let's add a new one ?... 

I think I understand the rest: the user specifies the network to which the pool applies when the pool is created. 

Another question: by `network` here, do you actually mean NAD ?